iavl/iavl_tree.go

```diff
@@ -1,37 +1,131 @@
+/*
+	Package merkle implements a key/value store with an immutable Merkle tree.
+*/
+
 package iavl
 
 import (
 	"bytes"
 	"container/list"
+	"fmt"
 	"sync"
 
 	wire "github.com/tendermint/go-wire"
-	. "github.com/tendermint/tmlibs/common"
-	dbm "github.com/tendermint/tmlibs/db"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/tendermint/tmlibs/db"
 	"github.com/tendermint/tmlibs/merkle"
 )
 
+// Keys for administrative persistent data
+var rootsKey = []byte("go-merkle.roots")      // Database key for the list of roots
```

Comment:
I wonder if we should consolidate  these down to a single struct, at least for things that are always saved and loaded together (version, root, orphans). Maybe the deletes stay separate as they can get pruned and saved in the background.  Is there ever a reason to load version/root/orphans independently of the others? Otherwise we can save a bunch of serialization code.

```
type TreeState struct{
  Version int
  Roots [][]byte
  Orphans [][]byte
}
```
Of course then we'd need to have funcs for mapping the list.List to [][]byte and back ...